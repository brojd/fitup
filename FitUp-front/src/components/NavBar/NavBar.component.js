import React, { PropTypes, Component } from 'react';
import { Link } from 'react-router';
import classNames from 'classnames';
import styles from './NavBar.stylesheet.css';

class NavBar extends Component {
  
  markAsVisited(elem) {
    let allNavAnchors = document.querySelectorAll(`.${styles.navList__anchor}`);
    for (let anchor of allNavAnchors) {
      anchor.className = styles.navList__anchor;
    }
    elem.target.className = `${styles.navList__anchor} ${styles.currentTab}`;
  }
  
  render() {
    return (
      <nav className={classNames('uk-width-10-10',  styles.NavBar)}>
        <ul className={classNames('uk-navbar-nav uk-hidden-small uk-align-center', styles.navList, this.props.className)}>
          <li className={styles.navList__elem}>
            <Link to='/'
                  className={`${styles.navList__anchor}`}
                  onClick={this.markAsVisited}>
              schedule
            </Link>
          </li>
          <li className={styles.navList__elem}>
            <Link to='gyms'
                  className={styles.navList__anchor}
                  onClick={this.markAsVisited}>
              gyms
            </Link>
          </li>
          <li className={styles.navList__elem}>
            <Link to='trainers'
                  className={styles.navList__anchor}
                  onClick={this.markAsVisited}>
              trainers
            </Link>
          </li>
          <li className={styles.navList__elem}>
            <Link to='bmi'
                  className={styles.navList__anchor}
                  onClick={this.markAsVisited}>
              BMI calculator
            </Link>
          </li>
        </ul>
      </nav>
    );
  }
}

NavBar.propTypes = {
  className: PropTypes.string
};

export default NavBar;
