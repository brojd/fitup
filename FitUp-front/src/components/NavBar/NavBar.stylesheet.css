.NavBar {
  background: #000000;
  height: 80px;
  display: flex;
  align-items: center;
  position: relative; }

.navList {
  margin-bottom: 0 !important; }

.navList__elem .navList__anchor {
  color: #FFFFFF;
  background: #000000;
  text-shadow: 0px 0px 6px #000000, 1px 1px 4px #8a0000, -1px -1px 4px #8a0000; }

.navList__elem:not(:first-of-type) {
  margin-left: 100px; }

.navList__elem:hover > .navList__anchor {
  color: #ffdede;
  cursor: pointer;
  animation: pulse 0.3s forwards;
  border-radius: 5%;
  background: rgba(255, 255, 255, 0.19); }

.navList__anchor:focus, .navList__anchor:active {
  background: rgba(255, 255, 255, 0.19);
  color: #ffdede; }

.currentTab {
  background: linear-gradient(to bottom right, #690000, #980000) !important;
  color: #FFFFFF !important;
  border-radius: 4px; }

@keyframes pulse {
  from {
    transform: scale3d(1, 1, 1); }
  to {
    transform: scale3d(1.05, 1.05, 1.05); } }

/*# sourceMappingURL=NavBar.stylesheet.css.map */
