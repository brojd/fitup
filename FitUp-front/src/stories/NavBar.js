import React from 'react';
import { storiesOf} from '@kadira/storybook';
import NavBar from '../components/NavBar/NavBar.component';

storiesOf('NavBar', module)
  .add('header navigation bar', () => (
    <NavBar />
  ));
