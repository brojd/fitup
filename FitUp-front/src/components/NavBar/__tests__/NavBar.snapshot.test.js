import React from 'react';
import NavBar from '../NavBar.component';
import renderer from 'react-test-renderer';

test('NavBar renders correctly', () => {
  const component = renderer.create(
    <NavBar className='someClass' />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
