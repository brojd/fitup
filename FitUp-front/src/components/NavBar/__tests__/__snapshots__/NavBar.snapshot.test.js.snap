exports[`test NavBar renders correctly 1`] = `
<nav
  className="uk-width-10-10 NavBar">
  <ul
    className="uk-navbar-nav uk-hidden-small uk-align-center navList someClass">
    <li
      className="navList__elem">
      <a
        className="navList__anchor"
        onClick={[Function]}
        style={Object {}}>
        schedule
      </a>
    </li>
    <li
      className="navList__elem">
      <a
        className="navList__anchor"
        onClick={[Function]}
        style={Object {}}>
        gyms
      </a>
    </li>
    <li
      className="navList__elem">
      <a
        className="navList__anchor"
        onClick={[Function]}
        style={Object {}}>
        trainers
      </a>
    </li>
    <li
      className="navList__elem">
      <a
        className="navList__anchor"
        onClick={[Function]}
        style={Object {}}>
        BMI calculator
      </a>
    </li>
  </ul>
</nav>
`;
